### PR TITLE
feat(ota): improve tests and make the progress a stream

### PR DESCRIPTION
**Pull request by joshuachp**
Remove the delays and channels from the tests. Replace the test that require to receive a channel as "expect" and make it return a progress stream directly.

### CODE REVIEW

File: src/ota/ota_handler.rs [hunk 1]
```diff
@@ -142,18 +142,27 @@ impl OtaHandler {
...
-            match operation_str.parse::<OtaOperation>() {
-                Ok(OtaOperation::Update) => self.handle_update(sdk, data).await,
-                Ok(OtaOperation::Cancel) => self.handle_cancel(sdk, data).await,
-                Err(_) => Err(DeviceManagerError::OtaError(OtaError::Request(
+        let operation_str = match data.get("operation") {
+            Some(AstarteType::String(operation_str)) => operation_str,
+            _ => {
+                error!("missing ota operation: {:?}", data);
+
+                return Err(DeviceManagerError::OtaError(OtaError::Request(
```
Comment (harlem88) [inline src/ota/ota_handler.rs:150]: Here can use `let-else`
```suggestion
        let Some(AstarteType::String(operation_str)) = data.get("operation") else{
                error!("missing ota operation: {:?}", data);
                return Err(DeviceManagerError::OtaError(OtaError::Request(
```

File: src/ota/ota_handler.rs [hunk 2]
```diff
@@ -162,43 +171,62 @@ impl OtaHandler {
...
-
-            self.sender.send(msg).await.map_err(|_| {
-                DeviceManagerError::OtaError(OtaError::Internal(
-                    "Unable to execute HandleOtaEvent, receiver channel dropped",
-                ))
-            })?;
+        let operation_str = match data.get("uuid") {
+            Some(AstarteType::String(operation_str)) => operation_str,
+            _ => {
+                error!("update data missing uuid: {:?}", data);
```
Comment (harlem88) [inline src/ota/ota_handler.rs:177]: same above
```suggestion
        let Some(AstarteType::String(operation_str) = data.get("uuid") else {
                error!("update data missing uuid: {:?}", data);
                return Ok(());
          }
```

File: src/ota/ota_handler.rs [hunk 3]
```diff
@@ -239,84 +267,86 @@ impl OtaHandler {
...
-                Ok(ota_status) => ota_status,
-                Err(err) => {
-                    let message = "Unable to cancel OTA request";
-                    error!("{message} : {err}");
-                    send_ota_event(
-                        sdk,
-                        &OtaStatus::Failure(OtaError::Internal(message), Some(cancel_ota_request)),
-                    )
-                    .await?;
+        let request_uuid_str = match &data.get("uuid") {
```
Comment (harlem88) [inline src/ota/ota_handler.rs:270]: same above use `let-else`

**Review by harlem88 — CHANGES_REQUESTED** [review]

File: src/ota/ota_handler.rs (hunk 1 above)
Comment (joshuachp) [reply src/ota/ota_handler.rs]: Done

File: src/ota/ota_handler.rs (hunk 2 above)
Comment (joshuachp) [reply src/ota/ota_handler.rs]: Done

File: src/ota/ota_handler.rs (hunk 3 above)
Comment (joshuachp) [reply src/ota/ota_handler.rs]: Done

**Review by rgallor — CHANGES_REQUESTED** [review]

File: src/ota/ota_handle.rs [hunk 4]
```diff
@@ -458,8 +477,6 @@ where
 
         info!("Rebooting in 5 seconds");
```
[the remark this answers is not in the document]
Comment (joshuachp) [reply src/ota/ota_handle.rs:478]: Yes, but misleading. I will change it

**Review by harlem88 — APPROVED** [review]

**Review by rgallor — APPROVED** [review]

**Pull request merged**